exporter/splunkhecexporter/client.go

```diff
@@ -194,14 +194,34 @@ func (c *client) pushLogRecords(ctx context.Context, lds plog.ResourceLogsSlice,
 		if state.bufFront == nil {
 			state.bufFront = &index{resource: state.resource, library: state.library, record: k}
 		}
+		var b []byte
+
+		if c.config.ExportRaw {
+			body := logs.At(k).Body().AsRaw()
+			if bodyAsBytes, ok := body.([]byte); ok {
+				b = bodyAsBytes
+			} else if bodyAsString, ok := body.(string); ok {
+				b = []byte(bodyAsString)
+			} else {
+				var err error
+				b, err = jsoniter.Marshal(body)
+				if err != nil {
+					permanentErrors = append(permanentErrors, consumererror.NewPermanent(fmt.Errorf("dropped raw: %v, error: %w", body, err)))
+					continue
+				}
+			}
+			b = append(b, '\n')
```

Comment:
I believe we can do just this instead

```suggestion
			b := []byte(logs.At(k).Body().AsString() + "\n")
```

Reply:
Thanks, that mostly does it, but bytes bodies get base64 encoded:
```
        	            	expected: "mybytes\n"
        	            	actual  : "bXlieXRlcw==\n"
```
So I might carve out an exception for bytes bodies.

Reply:
done - thanks!

Reply:
Is the Splunk Raw endpoint really supposed to receive _any_ bytes?

Reply:
I don’t know about any bytes as in bytes outside UTF-8 or weird encodings, and I am going to ask.

Reply:
I don't think we should send raw bytes as is. I don't know about any receiver that would encode a string into a log body of Bytes type.

Reply:
I think we should avoid using `AsString()` method here. It internally converts a byte array to a string for map and array. It will lead to additional memory read/write to append to string and convert it back to byte array.

Reply:
> I think we should avoid using AsString() method here. It internally converts a byte array to a string for map and array. It will lead to additional memory read/write to append to string and convert it back to byte array.

There is no way in any existing receivers to set the body of bytes type. It's either string or map. And what is the reason to send **any** byte sequence as is? That can be pretty dangerous. If UTF-8 encoding is assumed, I think we should send UTF-8.

Reply:
I reached out internally and was told base64 is fine. I think we can stick with that and folks can always open an issue if that starts to get in the way. Let's see people use this for a bit, it's ok.

Reply:
Yup, we can create a follow-up PR with benchmarks. In the benchmark.